Add department and initiative filters to statistics API

denarch22 opened the pull request:
## 🛠️ Changes
Introduces endpoints and service methods to retrieve general statistics filtered by department or initiative. Updates repository interfaces and implementations to support filtered queries for active initiatives, people involved, and area calculations.

## 📝 Associated issues
Resolve [LIB-351](https://linear.app/biodev/issue/LIB-351/monitoreo-comunitario-actualizar-controlador-de-cifras-generales)

denarch22 committed:
Add department and initiative filters to statistics API

Introduces endpoints and service methods to retrieve general statistics filtered by department or initiative. Updates repository interfaces and implementations to support filtered queries for active initiatives, people involved, and area calculations.

cgalvist commented on src/Application/Services/Statistics/GeneralStatisticsService.cs:
```diff
@@ -75,6 +75,172 @@ public async Task<CustomWebResponse> GetGeneralStatisticsAsync(CancellationToken
         };
     }
 
+    /// <summary>
```
Esto me parece código similar o duplicado de la función `GetGeneralStatisticsAsync(CancellationToken ct = default)` ¿sería posible integrar las dos funciones en una sola?

cgalvist commented on src/Application/Services/Statistics/GeneralStatisticsService.cs:
```diff
@@ -75,6 +75,172 @@ public async Task<CustomWebResponse> GetGeneralStatisticsAsync(CancellationToken
+    /// <param name="initiativeId">Initiative identifier.</param>
+    /// <param name="ct">Cancellation token.</param>
+    /// <returns>General statistics data for the initiative.</returns>
+    public async Task<CustomWebResponse> GetGeneralStatisticsByInitiativeAsync(int initiativeId, CancellationToken ct = default)
```
Tengo la misma opinión anterior. Creo que se puede unificar toda la lógica en una misma función

cgalvist commented on src/Application/Services/Statistics/GeneralStatisticsService.cs:
```diff
@@ -75,6 +75,172 @@ public async Task<CustomWebResponse> GetGeneralStatisticsAsync(CancellationToken
+    /// <param name="departmentId">Department identifier.</param>
+    /// <param name="ct">Cancellation token.</param>
+    /// <returns>Count of active initiatives in the department.</returns>
+    private async Task<int> GetTotalActiveInitiativesCountByDepartmentAsync(int departmentId, CancellationToken ct = default) =>
```
Considero que esta función, `GetTotalPeopleInvolvedCountByDepartmentAsync(int departmentId, CancellationToken ct = default)`, `GetTotalActiveInitiativesCountByInitiativeAsync(int initiativeId, CancellationToken ct = default)` y `GetTotalPeopleInvolvedCountByInitiativeAsync(int initiativeId, CancellationToken ct = default)` no son necesarias. Creo que pueden usar las funciones de los repositorios y ahorrar algunas líneas de código

denarch22 committed:
Refactor statistics service to unify logic

Consolidated general statistics retrieval into a single internal method with optional filters for department and initiative. Removed redundant private methods and simplified public API, improving maintainability and reducing code duplication.

cgalvist reviewed (changes requested):
Hola @denarch22.

Gracias por tus aportes. Considero que está bien, lo único que hace falta son los ejemplos de respuesta para Swagger de `GeneralStatisticsController`.

denarch22 committed:
Add Swagger response type documentation to GeneralStatisticsController
